common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -200,7 +203,11 @@ public Collation(
       // No Collation can simultaneously support binary equality and lowercase equality
       assert(!supportsBinaryEquality || !supportsLowercaseEquality);
 
```

Comment:
Please add comment why PROVIDER_NULL is now acceptable, in addition to SUPPORTED_PROVIDERS